lib/src/arithmetic/divider.dart

```diff
@@ -365,16 +368,17 @@ class MultiCycleDivider extends Module {
         ElseIf(currentState.eq(_MultiCycleDividerState.ready) & intf.validIn, [
           lastSuccess < 0,
           lastDifference <
-              mux(intf.dividend[dataWidth - 1] & intf.isSigned,
-                  ~intf.dividend + 1, intf.dividend), // start by matching aBuf
+              mux(extDividendIn[dataWidth - 1] & intf.isSigned,
+                  ~extDividendIn + 1, extDividendIn), // start by matching aBuf
         ]),
         ElseIf(
             currentState.eq(_MultiCycleDividerState
                 .process), // didn't exceed a_buf, so count as success
             [
               If(~tmpDifference[dataWidth - 1], then: [
                 lastSuccess <
-                    (Const(1, width: dataWidth) << currIndex), // capture 2^i
```

Comment:
Should we make a feature request for width inference on muxes?  I even created a 'zero' and 'one' variable once to help minimize my typing with muxes, but I also ran into a lot of bugs when forgetting to expand a 0 or a 1 to the correct width.

Reply:
I like that idea! That's often the source of build errors for me - forgetting to set the width correctly on a Const.